pandas/core/dimensioned.py

```diff
@@ -0,0 +1,245 @@
+import numpy as np
+
+from pandas.core.base import (PandasObject)
+from pandas.util._decorators import cache_readonly
+from pandas import compat
+from pandas.core.common import is_null_slice
+
+
+class Dimensional(PandasObject):
+    """
```

Comment:
I tried that. But I didn't get it to work properly. I guess I don't know enough about the internals of pandas to get it to work.